src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -442,6 +444,84 @@ struct aie_reg_read
   }
 };
 
+struct aim_counter
+{
+  using result_type = query::aim_counter::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
```

Comment:
You are and rightly so.  This is the pattern for implementing the query request getters you see in query_request.h.  You will have to follow a trail of template instantiation to get this this get() function.  In this case the get function has custom implementation, in other cases it will translate into a sysfs call.